pkg/cmd/server/start/start_master.go

```diff
@@ -102,23 +103,28 @@ func NewCommandStartMaster(out io.Writer) (*cobra.Command, *MasterOptions) {
 	}
 	cmd.SetOutput(out)
 
-	options.MasterArgs = NewDefaultMasterArgs()
-
 	flags := cmd.Flags()
+	options.MasterArgs = MasterArgsAndFlags(flags)
 
 	flags.Var(options.MasterArgs.ConfigDir, "write-config", "Directory to write an initial config into.  After writing, exit without starting the server.")
 	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from. When running from a configuration file, all other command-line arguments are ignored.")
 	flags.BoolVar(&options.CreateCertificates, "create-certs", true, "Indicates whether missing certs should be created")
 
-	BindMasterArgs(options.MasterArgs, flags, "")
-	BindListenArg(options.MasterArgs.ListenArg, flags, "")
-	BindImageFormatArgs(options.MasterArgs.ImageFormatArgs, flags, "")
-	BindKubeConnectionArgs(options.MasterArgs.KubeConnectionArgs, flags, "")
-	BindNetworkArgs(options.MasterArgs.NetworkArgs, flags, "")
-
 	return cmd, options
 }
 
+func MasterArgsAndFlags(flags *pflag.FlagSet) *MasterArgs {
```

Comment:
I think this should be useful even in a self-compiled run-with-options setup. It may not seem like there's much to do with a master or node from this setup, since you have no config to check and probably no logs (unless we introduce an option to specify where to find those and scrape them in addition to journald) - but you can at least check systemd dependencies like Docker, possibly etcd or kubernetes, which could be useful.